api/analyzers/source_analyzer.py

```
@@ -81,61 +111,61 @@
-            self.first_pass(ignore, executor)
+    def analyze_sources(self, path: Path, ignore: list[str], graph: Graph) -> None:
+        # First pass analysis of the source code
+        self.first_pass(path, ignore, graph)
 
-            # Second pass analysis of the source code
-            self.second_pass(ignore, executor)
+        # Second pass analysis of the source code
+        self.second_pass(graph, path)
 
-    def analyze_local_folder(self, path: str, g: Graph, ignore: Optional[List[str]] = []) -> None:
+    def analyze_local_folder(self, path: str, g: Graph, ignore: Optional[list[str]] = []) -> None:
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)